FRFuel.cs

```diff
@@ -164,32 +166,62 @@ public void CreateBlips()
         }
 
         /// <summary>
-        /// Creates jerry cans pick-ups at the gas stations
+        /// Gets coordinates for jerry cans within 100f radius
         /// </summary>
-        public void CreateJerryCanPickUps()
+        /// <param name="position"></param>
+        /// <returns>List of coordinates for pickups</returns>
+        public IEnumerable<Vector3> GetNearbyJerryCanPickUpCoordinates(Vector3 position)
+        {
+            return GasStations.positions.Where(p => p.DistanceToSquared(position) < 100.0f);
+        }
+
+        /// <summary>
+        /// Automatically adds pickups for nearby jerry cans, and removes when leaving area
+        /// </summary>
+        public async Task ManageNearbyJerryCanPickUps()
         {
             if (Config.Get("CreatePickups", "true") != "true")
             {
                 return;
             }
 
+            Vector3 pos = Function.Call<Vector3>(Hash.GET_ENTITY_COORDS, Function.Call<int>(Hash.PLAYER_PED_ID));
+
             int model = 883325847;
 
             Function.Call(Hash.REQUEST_MODEL, model);
 
-            for (int i = 0; i < GasStations.positions.Length; i++)
+            while (Function.Call<bool>(Hash.HAS_MODEL_LOADED, model))
```

Comment:
shouldn't this be `!` loaded instead? seems like an infinite loop would happen here.